### ISSUE TITLE
multiple app icons on macOS

### ISSUE THREAD

**Issue by dae** [ankitects/anki#1812]
https://forums.ankiweb.net/t/double-icons-in-dock-of-mac-os-monterey-intel/19148

I see a brief expansion of the dock when opening (as if an icon is about to be added there), but it is momentary and the dock shrinks back down again afterwards. I can't seem to reproduce the issue the others are experiencing, and don't know what's causing this. If anyone has any ideas, please let us know!

To test with a built app, you can run tools/bundle, then look in .bazel/out/build/app.

**Comment by dumduma** [comment]
I can reproduce this on Big Sur 11.04 with m1 chip.  

>I see a brief expansion of the dock when opening (as if an icon is about to be added there), but it is momentary and the dock shrinks back down again afterwards. 

Do you have "Show recent applications in Dock" turned on? It can be found in System Preferences > Dock & Menu Bar

**Comment by dae** [comment]
I see the brief animation whether that's turned on or off, and don't see the duplicate icons in either case.

**Comment by shenlebantongying** [comment]
I have two machines:

On qt5 intel macOS 11.6 "Big Sur": No problem
On qt5 & qt6 M1 macOS 12.3.1 "Monterey": Duplicating recent apps

This may be a system Bug? (or a Feature?)

**Comment by shenlebantongying** [comment]
I am on M1 macOS Monterey 12.3.1

This problem is gone if I use the (Intel, 2.1.49) -> https://github.com/ankitects/anki/releases/download/2.1.49/anki-2.1.49-mac.dmg

The icons will keep duplicating with (Apple Silicon, 2.1.51 Qt6) https://github.com/ankitects/anki/releases/download/2.1.51/anki-2.1.51-mac-apple-qt6.dmg

**Comment by shenlebantongying** [comment]
I used a hack by adding those lines to `/Applications/Anki.app/Contents/Info.plist` and the duplicating problem is gone.

```
<key>LSUIElement</key>
<string>1</string>
```

I don't know the theory, i googled it -> https://stackoverflow.com/a/17174432/12494649

**Comment by dae** [comment]
@shenlebantongying you deleted your comments; did your solution not work?

**Comment by shenlebantongying** [comment]
I tried anki ⁨2.1.51 (cf78a555)⁩ Python 3.9.7 Qt 6.2.3 PyQt 6.2.3 on two machines:

macOS 11.6, Big Sur -> Intel -> As you described, the "brief expansion of the dock when opening"
macOS 12.3.1 Monterey -> M1   -> Duplicating.

It may be a macOS's bug. Maybe we should ask the system version in the discourse to confirm this.

**Comment by shenlebantongying** [comment]
Some pokings which I did and I think they are useless:

No, it doesn't work, it just transforms it into another bug -> there will be one icon, but the icon on the dock doesn't associate with Anki anymore, and there will be no menubar. 

I added this to anki.app's plist. https://developer.apple.com/documentation/bundleresources/information_property_list/lsuielement
```
<keay>LSUIElement</key>
<string>1</string>
```

My theory is that macOS consider GUI app started by "direct exec()/system()" within an application as another app. I guess anki internally spawns the PyQt instance somehow mac doesn't like it.

I am not sure why you cannot reproduce it, but do u open it in a developing environment like open it through the command line?

Also, vscode suffered a similar issue before, this one same reasonable,https://github.com/microsoft/vscode/issues/60579#issuecomment-554652633

**Comment by shenlebantongying** [comment]
To reproduce this bug, you have to enable "Show recent apps in dock" at Preference -> Dock & Menu Bar

**Comment by dae** [comment]
Ok, I was able to reproduce it. The things I was missing:

- the .app must be in /Applications
- if you have more than three running applications in the recent list already, you won't see any new ones, so I had to close some running apps to trigger the behaviour

It turns out to be caused by mpv. When MpvManager() is called in setup_sound() in main.py, an extra dock icon is created. I tried passing --macos-app-activation-policy=accessory or prohibited to mpv, but it did not help. Ideas welcome :-)
